pkg/auth/auth.go

```diff
@@ -55,12 +95,15 @@ func Extract(ctx context.Context) *User {
 // Returning the new context that has been created.
 func ToContext(ctx context.Context, u *User) context.Context {
 	if u == nil || u.Email == "" {
-		u = DefaultUser
+		u = MakeDefaultUser()
 	}
 	// if no username was specified, use email as username
 	if u.Name == "" {
 		u.Name = u.Email
 	}
+
+	u.Name = encode64(u.Name)
+	u.Email = encode64(u.Email)
 	ctx = metadata.AppendToOutgoingContext(ctx, "author-email", u.Email, "author-username", u.Name)
```

Comment:
The grpc standard says you need to suffix binary headers with "-bin". Probably grpc-go is doing something else then: https://github.com/grpc/grpc-go/blob/master/Documentation/grpc-metadata.md#storing-binary-data-in-metadata

Reply:
I guess that should work too. But now that I have a working solution, I'm a bit hesitant to try out another way.